esx_service/vmdk_ops.py

```diff
@@ -854,11 +854,14 @@ def findDeviceByPath(vmdk_path, vm):
         # to match with the given volume name.
         # Filename format is as follows:
         #   "[<datastore name>] <parent-directory>/tenant/<vmdk-descriptor-name>"
-        logging.debug("d.backing.fileName %s", d.backing.fileName)
-        backing_disk = d.backing.fileName.split(" ")[1]
+        # Assume there is one space between these two parts "[<datastore name>]" and
+        # <parent-directory>/tenant/<vmdk-descriptor-name>
+        logging.debug("findDeviceByPath: d.backing.fileName %s", d.backing.fileName)
+        backing_disk = d.backing.fileName.rsplit(" ", 1)[1]
```

Comment:
Can  `<parent-directory>/tenant/<vmdk-descriptor-name>` hasve a space ? The current code is guaranteed to fail. Maybe it's better to look for ']' ? Something like
```python
    ds, file  = d.backing.fileName.rsplit("]", 1)[1]
    datastore = ds[1:]
    backing_disk = file[1:]
```
That raises a couple of additional things to check:
(1) can we create a tenant with a space 
(2) can we create a volume with a space in the name.